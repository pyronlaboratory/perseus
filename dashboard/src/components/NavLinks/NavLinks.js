import React from 'react'
import clsx from 'clsx';

// react router
import {Link, withRouter } from "react-router-dom";

// material imports
import { makeStyles } from '@material-ui/core/styles';

import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';

// material icons
import ListItemIcon from '@material-ui/core/ListItemIcon';

const useStyles = makeStyles((theme) => ({

    item: {
      paddingTop: 8,
      paddingBottom: 8,
      color: "#343434",
      '&:hover,&:focus': {
        backgroundColor: "#f3f3f3", 
      },
    },
    itemActiveItem: {
      color: '#4fc3f7',
    },
    itemPrimary: {
      fontSize: 'inherit',
    },
    itemIcon: {
      minWidth: 'auto',
      marginRight: theme.spacing(2),
    },
    divider: {
      marginTop: theme.spacing(2),
    },
  
}));

function NavLinks(props) {

    const classes = useStyles();

    return (
        <Link key={props.id} to={props.item.url}>
        <ListItem
            button
            className={clsx(classes.item, props.item.url === props.history.location.pathname && classes.itemActiveItem)}
        >
            <ListItemIcon className={classes.itemIcon}>{props.item.icon}</ListItemIcon>
            <ListItemText
                classes={{
                primary: classes.itemPrimary,
                }}
            >
                {props.item.id}
            </ListItemText> 
        </ListItem>
      </Link>
    )
}

export default withRouter(NavLinks);