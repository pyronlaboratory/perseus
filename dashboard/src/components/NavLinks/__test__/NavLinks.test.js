import React from 'react';
import ReactDOM from 'react-dom';

// react testing
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import NavLinks from './../NavLinks';
import { BrowserRouter as Router } from "react-router-dom";

afterEach(cleanup);

describe('nav link component', () => {
    it("renders without crashing", () => {
        const div = document.createElement("div");
        const mockItem = {
            id: "id",
            icon: "icon",
            url: "url"
        }
        ReactDOM.render(
            <Router>
                <NavLinks key={mockItem.id} item={mockItem}/> 
            </Router>
            , div);
    })

    it("matches snapshot", () => {
        const mockItem = {
            id: "id",
            icon: "icon",
            url: "url"
        }
        const tree = renderer.create(
            <Router>
                <NavLinks key={mockItem.id} item={mockItem}/> 
            </Router>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})