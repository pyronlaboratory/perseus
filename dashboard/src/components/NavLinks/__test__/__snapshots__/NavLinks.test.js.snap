// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav link component matches snapshot 1`] = `
<a
  href="/url"
  onClick={[Function]}
>
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root makeStyles-item-1 MuiListItem-gutters MuiListItem-button"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root makeStyles-itemIcon-4"
    >
      icon
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary makeStyles-itemPrimary-3 MuiTypography-body1 MuiTypography-displayBlock"
      >
        id
      </span>
    </div>
  </div>
</a>
`;
